op-program/host/host.go

```diff
@@ -24,8 +26,20 @@ type L2Source struct {
 	*sources.DebugClient
 }
 
+const opProgramChildEnvName = "OP_PROGRAM_CHILD"
+
+func RunningProgramInClient() bool {
+	value, _ := os.LookupEnv(opProgramChildEnvName)
+	return value == "true"
+}
+
 // FaultProofProgram is the programmatic entry-point for the fault proof program
 func FaultProofProgram(logger log.Logger, cfg *config.Config) error {
+	if RunningProgramInClient() {
+		cl.Main(logger)
+		// unreachable
```

Comment:
```suggestion
		panic("Client main should have exited process")
```
Just because everything unreachable one day gets reached and a clear panic will be so much easier to debug. :)